test/e2e/data/autoscaler/autoscaler-to-workload-workload.yaml

```diff
@@ -0,0 +1,202 @@
+# This yaml deploys the autoscaler on a workload cluster and configures it to match
+# against the corresponding Cluster API cluster which is defined into the management cluster.
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cluster-autoscaler-system
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler
+  namespace: cluster-autoscaler-system
+  labels:
+    app: cluster-autoscaler
+spec:
+  selector:
+    matchLabels:
+      app: cluster-autoscaler
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cluster-autoscaler
+    spec:
+      containers:
+        - image: k8s.gcr.io/autoscaling/cluster-autoscaler:${AUTOSCALER_VERSION}
+          name: cluster-autoscaler
+          command:
+            - /cluster-autoscaler
+          args:
+            - --cloud-provider=clusterapi
+            # Specify kubeconfig for management cluster
+            - --cloud-config=/management-cluster/kubeconfig
+            # Limit cluster autoscaler to only match against resources belonging to a single Cluster API cluster
+            - --node-group-auto-discovery=clusterapi:namespace=${CLUSTER_NAMESPACE},clusterName=${CLUSTER_NAME}
+          volumeMounts:
+            - name: kubeconfig-management-cluster
```

Comment:
It looks like this is following the pattern of installing the cluster-autoscaler deployment onto the mgmt cluster (as opposed to it being installed onto the workload cluster). Should we do a per-cluster `cluster-autoscaler` deployment namespace so that there is a path forward for managing multiple clusters?